pkg/controllers/machinesync/machine_sync_controller.go

```diff
@@ -172,7 +168,8 @@ func (r *MachineSyncReconciler) Reconcile(ctx context.Context, req reconcile.Req
 		logger.Info("machine currently migrating", "machine", mapiMachine.GetName())
 		return ctrl.Result{}, nil
 	default:
-		return ctrl.Result{}, fmt.Errorf("%w: %s", errAuthoritativeAPIUnexpected, mapiMachine.Spec.AuthoritativeAPI)
+		logger.Info("machine AuthoritativeAPI has unexpected value", "AuthoritativeAPI", mapiMachine.Status.AuthoritativeAPI, "machine", mapiMachine.GetName())
```

Comment:
self-nit: I don't think I need the extra machine key here, as it should be in the logging context.... I tihnk. 

Leaving here as unsure.

Reply:
You have the request name and namespace, the `machine` value will be a duplicate of the `name` right?

Reply:
yep ill remove the additional key then